File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.cs
public class RandomizedSet {
    
    HashSet<int> hash;

    public RandomizedSet() {
        hash = new HashSet<int>();
    }
    
    public bool Insert(int val) {
        if(hash.Contains(val))
            return false;
        hash.Add(val);
        return true;
    }
    
    public bool Remove(int val) {
        if(!hash.Contains(val))
            return false;
        hash.Remove(val);
        return true;
    }
    
    public int GetRandom() {
        int random = new Random().Next(hash.Count);
        int i = 0;
        foreach(int x in hash)
        {
            if(i == random)
                return x;   
            i++;
        }
        
        return 0;
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * bool param_1 = obj.Insert(val);
 * bool param_2 = obj.Remove(val);
 * int param_3 = obj.GetRandom();
 */

File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.cpp
class RandomizedSet {
public:
    
    unordered_set<int> temp;

    RandomizedSet() {
        
    }
    
    bool insert(int val) {
        if(temp.find(val) != temp.end())
            return false;
        temp.insert(val);
        return true;
    }
    
    bool remove(int val) {
        if(temp.find(val) != temp.end()) {
            temp.erase(val);
            return true;
        }
        return false;
     
    }
    
    int getRandom() {
        int n = temp.size();
        
        int random = rand() % n;
        int i=0; 
        for(auto x: temp) {
            if(i == random)
                return x;
            i++;
        }
        return 0;
        
        
    }
};

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet* obj = new RandomizedSet();
 * bool param_1 = obj->insert(val);
 * bool param_2 = obj->remove(val);
 * int param_3 = obj->getRandom();
 */